include/inexor/vulkan-renderer/wrapper/texture.hpp

```diff
@@ -81,6 +81,13 @@ class Texture {
     Texture &operator=(const Texture &) = delete;
     Texture &operator=(Texture &&) = default;
 
+    /// @brief Creates data for an error or a default texture.
+    /// @param width [out] The width of the created texture.
+    /// @param height [out] The height of the created texture.
+    /// @param channels [out] The number of channels in the created texture.
+    /// @return The pointer to the texture data.  The caller is responsible for freeing via delete[].
+    static unsigned char* load_error_texture_data(int* width, int* height, int* channels);
```

Comment:
In the future, I think this function should return a `Texture` directly, but as you said, `Texture` is currently a bit bulky so it's probably fine like this for now. Maybe you could put a TODO though.

Reply:
Not sure if i should agree or not.
- Out Parameter
- Returning a raw pointer in a public method
Those are already heavy reasons against it.

Reply:
This should be a lot better in the newest code with CpuTexture / GpuTexture classes.